sdcm/cluster.py

```diff
@@ -2745,9 +2746,15 @@ def get_gossip_info(self):
             if schema and ip and status:
                 if node := self.parent_cluster.find_node_by_ip(ip):
                     gossip_node_schemas[node] = {'schema': schema, 'status': status, 'dc': dc}
-                    schema = ip = status = dc = ''
                 else:
-                    LOGGER.error("Get gossip info. Failed to find a node in the cluster by IP: %s", ip)
+                    if status in self.GOSSIP_STATUSES_FILTER_OUT:
+                        LOGGER.debug("Get gossip info. Node with IP %s is not found in the cluster. Node status is: %s",
+                                     ip, status)
+                    else:
+                        LOGGER.error("Get gossip info. Failed to find a node with status %s in the cluster by IP: %s",
+                                     status, ip)
+
+                schema = ip = status = dc = ''
```

Comment:
instead of this, you could move the 1st time this line appears (on line 2735) inside the for-loop, so that one, and this one will be the same, or am i missing something?

Reply:
I added line 2735 to init the variables and prevent failure on non-existent variable (if the test wasn't found in the output).
Inside the loop the variables should be reset just when all of variable got value. And it doesn't happen every loop iteration.